sigstore/dsse/_predicate.py

```diff
@@ -0,0 +1,219 @@
+# Copyright 2024 The Sigstore Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Models for the predicates used in in-toto statements
+"""
+
+from typing import Any, Dict, List, Literal, Optional, TypeVar, Union
+
+from pydantic import (
+    BaseModel,
+    ConfigDict,
+    RootModel,
+    StrictBytes,
+    StrictStr,
+    model_validator,
+)
+from pydantic.alias_generators import to_camel
+
+from sigstore.dsse import Digest
+
+PREDICATE_TYPE_SLSA_v0_2 = "https://slsa.dev/provenance/v0.2"
+PREDICATE_TYPE_SLSA_v1_0 = "https://slsa.dev/provenance/v1"
+
+SUPPORTED_PREDICATE_TYPES = [PREDICATE_TYPE_SLSA_v0_2, PREDICATE_TYPE_SLSA_v1_0]
```

Comment:
Flagging for a follow-up: maybe we can make this an `enum.Enum` instead? That will reduce the number of unique imports we need to do, and allow us to define "supported" as "present in the enum".